### DISCUSSION TITLE
Pyright not working after installing with pip: SyntaxError: Unexpected token ;

### DISCUSSION THREAD

**Discussion by nathanlima** [microsoft/pyright#8217] [General]
Hello, everyone! I hope you all are doing well.

I'm trying to install pyright using `pip install pyright`. The installation succeed well, but when I type `pyright` in my prompt i'm getting the error below. I'm using ubuntu 20.04. Can someone help me to figure out what could I do to solve this issue? Thank you.

```
$ pyright
/home/myuser/.cache/pyright-python/1.1.368/node_modules/pyright/dist/vendor.js:2
exports.id=968,exports.ids=[968],exports.modules={6596:(A,e,t)=>{"use strict";const r=t(1160),o=global.Date;class i extends o{constructor(A){super(A),this.isDate=!0}toISOString(){return`${this.getUTCFullYear()}-${r(2,this.getUTCMonth()+1)}-${r(2,this.getUTCDate())}`}}A.exports=A=>{const e=new i(A);if(isNaN(e))throw new TypeError("Invalid Datetime");return e}},4023:(A,e,t)=>{"use strict";const r=t(1160);class o extends Date{constructor(A){super(A+"Z"),this.isFloating=!0}toISOString(){return`${this.getUTCFullYear()}-${r(2,this.getUTCMonth()+1)}-${r(2,this.getUTCDate())}T${r(2,this.getUTCHours())}:${r(2,this.getUTCMinutes())}:${r(2,this.getUTCSeconds())}.${r(3,this.getUTCMilliseconds())}`}}A.exports=A=>{const e=new o(A);if(isNaN(e))throw new TypeError("Invalid Datetime");return e}},692:A=>{"use strict";A.exports=A=>{const e=new Date(A);if(isNaN(e))throw new TypeError("Invalid Datetime");return e}},7640:(A,e,t)=>{"use strict"

SyntaxError: Unexpected token ;
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.o.f.require (/home/myuser/.cache/pyright-python/1.1.368/node_modules/pyright/dist/pyright.js:1:2393)
    at Promise.all.Object.keys.reduce (/home/myuser/.cache/pyright-python/1.1.368/node_modules/pyright/dist/pyright.js:1:1862)
    at Array.reduce (<anonymous>)
```

**Comment by erictraut** [comment]
This is probably because you have a version of node installed on your system that is too old to run pyright. Try updating to a recent version of node.

**Reply by MathanGeurtsen** [reply]
For reference, installing with latest lts works at the moment. (`nvm install --lts` currently points to npm 10.7.0 and node v20.15.0). 

Could version minimums be referenced somewhere in the installation documentation? 

I don't always have the liberty of choosing which versions to use, so being able to point to a version minimum can help upgrades happen.

**Reply by erictraut** [reply]
Pyright's documentation says "If you don't have a recent version of node on your system, install that first from [nodejs.org](https://nodejs.org)."

The latest version of node is v22. Pyright currently works as far back as v14. Versions of node older than v20 are no longer in active support, and versions older than v18 have no security support. It is therefore recommended that you at least have v18 (and preferably v20) installed.

**Reply by nathanlima** [reply]
Thank you @MathanGeurtsen and @erictraut , I followed what both of you suggested and this is working now. thank you!

**Reply by jxu** [reply]
I had the same problem - I'm on Ubuntu 22.04 and the node version in the repos is v12.
